TrafficCapture/dockerSolution/src/main/docker/migrationConsole/lib/console_link/console_link/models/backfill_base.py

```diff
@@ -69,5 +72,47 @@ def archive(self, *args, **kwargs) -> CommandResult[str]:
         Should fail if there are currently running operations."""
         pass
 
+    @abstractmethod
+    def build_backfill_status(self) -> BackfillStatus:
+        pass
+
     def describe(self) -> Dict:
         return self.config
+
+
+class BackfillOverallStatus(BaseModel):
+    status: StepState
+    percentage_completed: float
+    eta_ms: Optional[float] = None
+    started: Optional[datetime] = Field(
+        default=None,
+        description="Start time in ISO 8601 format",
+        json_schema_extra={"format": "date-time"}
+    )
+    finished: Optional[datetime] = Field(
+        default=None,
+        description="Start time in ISO 8601 format",
+        json_schema_extra={"format": "date-time"}
+    )
+    shard_total: Optional[int] = None
+    shard_complete: Optional[int] = None
+    shard_failed: Optional[int] = None
+    shard_in_progress: Optional[int] = None
+    shard_waiting: Optional[int] = None
```

Comment:
I know this isn't very granular with large shards not much progress can really be discovered, I've created https://opensearch.atlassian.net/browse/MIGRATIONS-2679 to make enhancements to how we track progress that would give us document counts / specific data metrics.